internal/backend/sessions/sessionworkflows/history.go

```diff
@@ -0,0 +1,385 @@
+		}
+
+		return sdktypes.NewStateSessionLogRecord(t, sdktypes.NewSessionStateStopped("[forced] "+a.WorkflowExecutionTerminatedEventAttributes.Reason)), nil
+
+	case *historypb.HistoryEvent_ActivityTaskScheduledEventAttributes:
+		a, ok := event.Attributes.(*historypb.HistoryEvent_ActivityTaskScheduledEventAttributes)
+		if !ok {
+			return sdktypes.InvalidSessionLogRecord, errors.New("not a scheduled activity event")
+		}
+
+		if a == nil {
+			l.Error("nil activity task scheduled event attributes from temporal")
+			return sdktypes.InvalidSessionLogRecord, nil
+		}
+
+		attrs := a.ActivityTaskScheduledEventAttributes
+		if attrs == nil {
+			l.Error("nil activity task scheduled event attributes from temporal")
+			return sdktypes.InvalidSessionLogRecord, nil
+		}
+
+		switch attrs.ActivityType.GetName() {
+		case sessioncalls.CallActivityName:
+			if types != 0 && types&sdktypes.CallSpecSessionLogRecordType == 0 {
+				return sdktypes.InvalidSessionLogRecord, nil
+			}
+
+			payloads := attrs.Input.GetPayloads()
+			if len(payloads) != 1 {
+				l.Error("unexpected number of payloads for activity task scheduled event", zap.Int("payloads", len(payloads)))
+				return sdktypes.InvalidSessionLogRecord, nil
+			}
+
+			var callInputs sessioncalls.CallActivityInputs
+			if err := json.Unmarshal(payloads[0].GetData(), &callInputs); err != nil {
+				return sdktypes.InvalidSessionLogRecord, temporalclient.TranslateError(err, "unmarshal activity input")
+			}
+
+			return sdktypes.NewCallSpecSessionLogRecord(t, callInputs.CallSpec), nil
+
+		case updateSessionStateActivityName:
+			if types != 0 && types&sdktypes.StateSessionLogRecordType == 0 {
+				return sdktypes.InvalidSessionLogRecord, nil
+			}
+
+			payloads := attrs.Input.GetPayloads()
+			if len(payloads) != 2 {
+				l.Error("unexpected number of payloads for activity task scheduled event", zap.Int("payloads", len(payloads)))
+				return sdktypes.InvalidSessionLogRecord, nil
+			}
+
+			var state sdktypes.SessionState
+			if err := json.Unmarshal(payloads[1].GetData(), &state); err != nil {
+				return sdktypes.InvalidSessionLogRecord, temporalclient.TranslateError(err, "unmarshal session state")
+			}
+
+			return sdktypes.NewStateSessionLogRecord(t, state), nil
+
+		case addSessionPrintActivityName:
+			if types != 0 && types&sdktypes.PrintSessionLogRecordType == 0 {
+				return sdktypes.InvalidSessionLogRecord, nil
+			}
+
+			payloads := attrs.Input.GetPayloads()
+			if len(payloads) != 3 {
+				l.Error("unexpected number of payloads for activity task print event", zap.Int("payloads", len(payloads)))
+				return sdktypes.InvalidSessionLogRecord, nil
+			}
+
+			var v sdktypes.Value
+			if err := v.UnmarshalJSON(payloads[1].GetData()); err != nil {
+				return sdktypes.InvalidSessionLogRecord, temporalclient.TranslateError(err, "unmarshal print text")
+			}
+
+			return sdktypes.NewPrintSessionLogRecord(t, v, 0), nil
+
+		default:
+			return sdktypes.InvalidSessionLogRecord, nil
```

Comment:
Log debugging data to warn us about the existence of this unexpected activity type

Reply:
no need - this is not unexpected,and not an error, just something we don't care about (yet?)